lib/utils.dart

```diff
@@ -0,0 +1,28 @@
+// Copyright 2018 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+String prettyPrintList(Iterable iter) {
+  if (iter == null) return 'NULL';
+
+  if (iter.length == 0) return '[]';
```

Comment:
Not needed. Your code below will handle the empty case fine.

Also, prefer `isEmpty`

Reply:
TIL about isEmpty. Fixed.